### PR TITLE
Converts lavaland /random mobs to landmarks

### PR DESCRIPTION
**Pull request by GDNgit**
## What Does This PR Do
Converts lavaland /random mobs to landmarks

## Why It's Good For The ~~Game~~ GC And Mappers
This serves the purpose of being landmarks while actually not being landmarks, saves an extremely small number of resources conditionally, ends up losing resources if we want to ever clear minds on GCing.  Converting to landmarks to fix this.
Also makes initialization a tiny bit faster

## Testing
Compiled, ran, you can guess the rest
## Changelog
NPFC

### CODE REVIEW

**Review by S34NW — CHANGES_REQUESTED** [review]
Instead of hardcoding it, perhaps you should set a `mob_type_to_spawn` variable or such and spawn that, with a prob override for the rare beasties

**Comment by GDNgit** [comment]
> Instead of hardcoding it, perhaps you should set a `mob_type_to_spawn` variable or such and spawn that, with a prob override for the rare beasties

Iffy due to it dealing with probs and it might cause OOO issues, but I'll try it out later

**Review by Sirryan2002 — CHANGES_REQUESTED** [review]
new'ing of mobs and the `return INITIALIZE_HINT_QDEL` should be handled in the base init
`/obj/effect/landmark/mob_spawner/Initialize(mapload)`

I like S34N's idea of the type to spawn being stored in a variable, this would allow you to pretty easily cutout a lot of duplicate code, you'd just need to pick the typepath before calling parent.

**Review by S34NW — CHANGES_REQUESTED** [review]
Please see sirryan's chanPlease see sirryan's changesges

**Comment by GDNgit** [comment]
Has been updated to work off parent

File: code/game/objects/effects/landmarks.dm [hunk 1]
```diff
@@ -588,6 +588,62 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/awaystart) //Without this away mission
 	ruin_template = null
 	. = ..()
 
+
+/// Mob spawners, spawns a mob and deletes the landmark
+
+/obj/effect/landmark/mob_spawner
```
Comment (SteelSlayer) [inline code/game/objects/effects/landmarks.dm:594]: The `"questionmark"`s below will be invisible unless you add `icon = 'icons/effects/spawner_icons.dmi'` here.

File: code/game/objects/effects/landmarks.dm [hunk 2]
```diff
@@ -588,6 +588,62 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/awaystart) //Without this away mission
...
+			return INITIALIZE_HINT_QDEL
+	else
+		. = ..()
+
+/obj/effect/landmark/mob_spawner/goldgrub
+	mobtype = /mob/living/simple_animal/hostile/asteroid/goldgrub
+	icon_state = "questionmark"
+
+/obj/effect/landmark/mob_spawner/goldgrub/Initialize(mapload)
+	. = ..()
```
Comment (SteelSlayer) [inline code/game/objects/effects/landmarks.dm:645]:
```suggestion
```

File: code/game/objects/effects/landmarks.dm [hunk 3]
```diff
@@ -588,6 +588,62 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/awaystart) //Without this away mission
...
 
+
+/// Mob spawners, spawns a mob and deletes the landmark
+
+/obj/effect/landmark/mob_spawner
+	///The mob we use for the spawner
+	var/mobtype = null
+
+/obj/effect/landmark/mob_spawner/Initialize(mapload)
+	. = ..()
```
Comment (SteelSlayer) [inline code/game/objects/effects/landmarks.dm:599]: I don't think these even need to call the parent since they're just going to be immediately deleted.

Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm]: Applied

File: code/game/objects/effects/landmarks.dm (hunk 2 above)
Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm]: Applied

File: code/game/objects/effects/landmarks.dm (hunk 1 above)
Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm:594]: Applied

File: code/game/objects/effects/landmarks.dm (hunk 3 above)
Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm]: linters doesn't play nice with this, even if it does pretty much nothing

File: code/game/objects/effects/landmarks.dm [hunk 4]
```diff
@@ -588,6 +588,60 @@ INITIALIZE_IMMEDIATE(/obj/effect/landmark/awaystart) //Without this away mission
...
+	return INITIALIZE_HINT_QDEL
+
+/obj/effect/landmark/mob_spawner/goliath
+	mobtype = /mob/living/simple_animal/hostile/asteroid/goliath/beast
+	icon_state = "questionmark"
+
+/obj/effect/landmark/mob_spawner/goliath/Initialize(mapload)
+	if(prob(1))
+		new /mob/living/simple_animal/hostile/asteroid/goliath/beast/ancient(loc)
+		return INITIALIZE_HINT_QDEL
```
Comment (Sirryan2002) [inline code/game/objects/effects/landmarks.dm:611]: for these `new`s, why not just set the `mobtype` equal to this path here and then let the parent function do all the heavy lifting? Then you wouldn't have to duplicate `return INITIALIZE_HINT_QDEL` which would cut down on code usage

Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm:611]: Damn you're right, lemme fix that in the next commit

Comment (GDNgit) [reply code/game/objects/effects/landmarks.dm]: Suggestion added

**Review by Sirryan2002 — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Comment by Sirryan2002** [comment]
This is a good refactor 👍 thank you

**Pull request merged**